src/models/CommonModel.ts

```diff
@@ -95,6 +96,7 @@ export class CommonModel extends CommonSchema<CommonModel> {
   addProperty(propertyName: string, propertyModel: CommonModel, schema: Schema) {
     if (this.properties === undefined) this.properties = {};
     if (this.properties[`${propertyName}`] !== undefined) {
+      Logger.warn(`While trying to add property to model, duplicate properties found. Merging models together for property ${propertyName}`, propertyModel, schema, this);
```

Comment:
Pleas ensure that there is a way to disable all log levels. Also from my point of view it should not be a `warn` but a `info`.

Reply:
The reason for choosing `warn` is that we are merging something that potentially can cause you to not get the output you expect. 

For example, say you have the following schema:
```
{
  "anyOf": [
    {
      "properties": {
        "test": {
          "$id": "test",
          ...
        }
      }
    },
    {
      "properties": {
        "test": {
          "$id": "test2",
          ...
        }
      }
    }
  ]
}
```

With the current merge functionality it results in the property `test` be a model with id `"test2"` and you might expect `"test"`. This is used consistently when something happens that you as a user might not expect.

Also in terms of disabling the logging, by default nothing is logged to the console, you have to set your own logging behavior. That reminds me that we need a doc for the logging behavior. Gonna add that.

Reply:
In my case I use allOf for merging those properties and it is something what I actually want.
But if you are saying that `warn` is better, let it be.

Reply:
I would say it is better to provide the user with a clear way (warn) of seeing what might do that you see an output you do not agree with.

This can of course be changed later if we figure out it is just in the way 🙂